### ISSUE TITLE
Unclear condition in shouldComponentUpdate in frontend/Props.js

### ISSUE THREAD

**Issue by tomaszlakomygp** [facebook/react-devtools#545]
This issue is a result of my findings in https://github.com/facebook/react-devtools/pull/535/files, @gaearon suggested to create a separate issue for that

Currently the `shouldComponentUpdate` function in https://github.com/facebook/react-devtools/blob/master/frontend/Props.js#L18 looks a bit uncommon:

```
  shouldComponentUpdate(nextProps: Object): boolean {
    return nextProps !== this.props;
  }
```

**Comment by gaearon** [comment]
This is probably unintentional. Worth verifying and fixing.

**Comment by ioantm** [comment]
There is one more place where this kind of check is done https://github.com/facebook/react-devtools/blob/master/frontend/Node.js#L44 , here `shouldComponentUpdate` probably should be removed.

**Comment by gaearon** [comment]
Fixed via #549.
